thunder/torch/__init__.py

```diff
@@ -1211,6 +1211,9 @@ def relu6(a: TensorProxy, /, inplace: bool = False) -> TensorLike:
 @torchsymbol(torch.nn.functional.hardswish, id="torch.hardswish", is_method=False)
 def hardswish(a: TensorProxy, /, inplace: bool = False) -> TensorLike:
     utils.check(not inplace, lambda: f"hardswish only supports inplace=False", exception_type=NotImplementedError)
+    utils.check(a.dtype in (torch.float16, torch.float32, torch.float64),
```

Comment:
This is a good update, and the PR is almost ready to be merged.

Two final tweaks:

1) This won't detect all floating point dtypes, like bfloat16, or future dtypes. Instead, use this function:

https://github.com/Lightning-AI/lightning-thunder/blob/56bc01b8e17ba9551395a23c639fae79bc9c25b3/thunder/core/dtypes.py#L440

2) The error message looks good, but don't set the `exception_type` or set it to `ValueError`. It's an interesting idea to consider dtype mismatch as a `TypeError`, but currently we don't consider the datatype of a tensor as part of its "type" (which is just tensor)